fabtests/include/unix/osd.h

```diff
@@ -34,6 +34,7 @@
 #define _FABTESTS_UNIX_OSD_H_
 
 #include <complex.h>
+#include <malloc.h>
```

Comment:
Maybe this was not the correct change. malloc.h is where mallopt is defined. Looks like mallopt is defined for Linux (and maybe other unix OSes) but not for osx. Since the osx build includes this header, maybe I should change this to:

```
#ifndef __APPLE__
#include <malloc.h>
#endif
```

Thoughts?

Reply:
It's best to avoid #ifdef's in code, as much as possible.  If malloc.h exists, it's okay to include it.  For osx, we can define mallopt() in the osd.h file

Reply:
malloc.h is needed on Linux for mallopt(). I thought that putting it in include/unix/osd.h may resolve the issue, but this header file is included on osx builds:

```
#ifdef __APPLE__
#include <osx/osd.h>
#include <unix/osd.h>
#elif defined __FreeBSD__
#include <freebsd/osd.h>
#include <unix/osd.h>
#elif defined _WIN32
#include <windows/osd.h>
#else
#include <unix/osd.h>
#endif
```

Note I have updated include/osx/osd.h with a mallopt() definition. But, this does not resolve the issue of the missing malloc.h. I need a way to not include this header with osx.

Reply:
Ah, I didn't realize that malloc.h doesn't exist on osx.  For libfabric, the way we handle missing headers is to define them (e.g. include/osx/malloc.h), and put the needed functions there (so define mallopt in osx/malloc.h).  See include/windows as an example.  I just don't think we've had this issue before with unix include files.

Reply:
Sounds good.

Reply:
I got the macos builds working. I had to update the fabtests configure.ac to include include/osx for macos.